io/net/bonding.py

```diff
@@ -0,0 +1,293 @@
+                  -f1" % val
+            cmd = "ssh %s@%s \"%s\"" % (self.user, self.peer_ip1, msg)
+            peer_ip = process.system_output(cmd, shell=True).strip()
+            cmd = 'echo %s | cut -d " " -f4' % peer_ip
+            peer_ip = process.system_output(cmd, shell=True).strip()
+            if peer_ip == "":
+                self.fail("test failed because peer ip can not retrieved")
+            self.peer_ips.append(peer_ip)
+        self.peer_interfaces.insert(0, self.peer_if1)
+        self.net_mask = []
+        st = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        for val in self.host_interfaces:
+            mask = socket.inet_ntoa(fcntl.ioctl(st.fileno(), 0x891b,
+                                                struct.pack('256s',
+                                                val))[20:24]).strip('\n')
+            self.net_mask.append(mask)
+        self.bonding_slave_file = "/sys/class/net/%s/bonding/slaves"\
+                                  % self.bond_name
+
+    def bond_remove(self, arg1):
+        '''
+        bond_remove
+        '''
+        if arg1 == "local":
+            self.log.info("Bonding configuration removed on laocal")
+            self.log.info("------------------------------------------------")
+            for ifs in self.host_interfaces:
+                cmd = "ifconfig %s down" % ifs
+                if process.system(cmd, shell=True, ignore_status=True) != 0:
+                    self.log.info("unable to bring down the interface")
+                cmd = "ifconfig %s down" % ifs
+                if process.system(cmd, shell=True, ignore_status=True) != 0:
+                    self.log.info("unable to bring down the interface")
+                cmd = "echo -%s > %s" % (ifs, self.bonding_slave_file)
+                if process.system(cmd, shell=True, ignore_status=True) != 0:
+                    self.log.info("bond removing failed in local machine")
+            cmd = "echo -%s > /sys/class/net/bonding_masters" % self.bond_name
+            if process.system(cmd, shell=True, ignore_status=True) != 0:
+                self.log.info("bond removing command failed in local machine")
+            time.sleep(5)
+        else:
+            self.log.info("Bonding configuration removed on Peer machine")
+            self.log.info("------------------------------------------------")
+            cmd = ''
+            cmd += 'ifconfig %s down;' % self.bond_name
+            for val in self.peer_interfaces:
+                cmd += 'ifconfig %s down;' % val
+            for val in self.peer_interfaces:
+                cmd += 'ip addr flush dev %s;' % val
+            for val in self.peer_interfaces:
+                cmd += 'echo "-%s" > %s;' % (val, self.bonding_slave_file)
+            cmd += 'echo "-%s" > /sys/class/net/bonding_masters;'\
+                   % self.bond_name
+            cmd += 'ifconfig %s %s netmask %s up;sleep 5;'\
```

Comment:
here also please use ifup as :

```
- An "ifconfig" command and "ifup" or "ifdown" command can activate network interfaces. "ifconfig" directly controls network interfaces, however "ifup" or "ifdown" executes "/etc/sysconfig/network-scripts/ifup-" or "ifdown-" scripts.
-  An "ifconfig up eth0" activates eth0 but does not setup IP addresses, however an "ifup eth0" setup IP addresses or other options based by an ifcfg-eth0, because the "ifup" uses an "ifup-eth*" script.
```

-